bluepysnap/edges.py

```diff
@@ -226,6 +569,35 @@ def _get(self, selection, properties=None):
 
         return result
 
+    def ids(self, edge_ids):
+        """Edge IDs corresponding to edges ``edge_ids``.
+
+        Args:
+            edge_ids (int/CircuitEdgeId/CircuitEdgeIds/sequence): Which IDs will be
+                returned depends on the type of the ``group`` argument:
+
+                - ``int``, ``CircuitEdgeId``: return a single edge ID.
+                - ``CircuitEdgeIds`` return IDs of edges in an array.
+                - ``sequence``: return IDs of edges in an array.
+
+        Returns:
+            numpy.array: A numpy array of IDs.
+        """
+        if isinstance(edge_ids, CircuitEdgeIds):
+            result = edge_ids.filter_population(self.name).get_ids()
+        elif isinstance(edge_ids, np.ndarray):
+            result = edge_ids
+        else:
+            result = utils.ensure_list(edge_ids)
+            # test if first value is a CircuitEdgeId if yes then all values must be CircuitEdgeId
+            if isinstance(next(iter(result), None), CircuitEdgeId):
```

Comment:
You want some lol ?

https://more-itertools.readthedocs.io/en/stable/api.html#more_itertools.first

```suggestion
            if isinstance(first(result, None), CircuitEdgeId):
```

Reply:
You convinced me about more-itertools ! It is already in the setup.